# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/new-regexp/regexp-parser.cc
# Commit: f53110858cb9
# Full Hash: f53110858cb9228ea0c7e056c3f7bc01b5fd76c9
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-05-12 04:04:10
# Regressor Bug: 1634135
# File Overlap Count: 1
# Description:
#   Bug 1634135: Disable named capture parsing until fully supported r=mgaudet
#   
#   The engine supports parsing named captures, but we don't have the code in place to expose the captured groups. Until we do, this code will make sure that we get a syntax error when parsing them.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73115
# ==============================================================================

diff -r 49b9d6a8a3a5 -r f53110858cb9 js/src/new-regexp/regexp-parser.cc
--- a/js/src/new-regexp/regexp-parser.cc	Sun May 10 16:00:25 2020 +0000
+++ b/js/src/new-regexp/regexp-parser.cc	Sun May 10 16:00:25 2020 +0000
@@ -1012,9 +1012,9 @@
                                     capture->name()->size());
     // CSA code in ConstructNewResultFromMatchInfo requires these strings to be
     // internalized so they can be used as property names in the 'exec' results.
-    Handle<String> name = factory->InternalizeString(capture_name);
-    array->set(i * 2, *name);
-    array->set(i * 2 + 1, Smi::FromInt(capture->index()));
+    // Handle<String> name = factory->InternalizeString(capture_name);
+    // array->set(i * 2, *name);
+    // array->set(i * 2 + 1, Smi::FromInt(capture->index()));
 
     i++;
   }